www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxBusyBufferConstraint.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/*
+* Copyright (C) 2019 Fabian Franz
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice,
+*    this list of conditions and the following disclaimer.
+*
+* 2. Redistributions in binary form must reproduce the above copyright
+*    notice, this list of conditions and the following disclaimer in the
+*    documentation and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+* INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+* AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+* OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+namespace OPNsense\Base\Constraints;
+
+
+use \Phalcon\Validation\Message;
+
+/**
+ * a very specific nginx check - not reusable
+ *
+ * Class NgxBusyBufferConstraint
+ * @package OPNsense\Nginx\Constraints
+ */
+class NgxBusyBufferConstraint extends BaseConstraint
```

Comment:
Clever solution 👍

Reply:
Thanks - seems to be the only usable callback for model validation.

The problem here is that the minimum and maximum values have to be calculated and depend on other values. What I hated most about that is that it was not in the online docs but raised an emerg error in the nginx config.

Reply:
@fabianfrz you can also bootstrap custom types per app model for more complex validations, like https://github.com/opnsense/core/blob/98673a0b30b48d68cee9bb0f52e0c68d1d6de51d/src/opnsense/mvc/app/models/OPNsense/Firewall/Alias.xml#L49

Reply:
@AdSchellevis I looked for that and did not find that. I think this should also go to the docs.

Reply:
that would be practical indeed